satpy/readers/agri_l1.py

```diff
@@ -0,0 +1,206 @@
+            # the value of dn is the index of brightness_temperature
+            data = self.calibrate(data, lut)
+            ds_info['valid_range'] = lut.attrs['valid_range']
+
+        data.attrs.update({'platform_name': self['/attr/Satellite Name'],
+                           'sensor': self['/attr/Sensor Identification Code'],
+                           'satellite_nominal_latitude': self['/attr/NOMCenterLat'],
+                           'satellite_nominal_longitude': self['/attr/NOMCenterLon'],
+                           'satellite_nominal_altitude': self['/attr/NOMSatHeight']})
+        data.attrs.update(ds_info)
+
+        # remove attributes that could be confusing later
+        data.attrs.pop('FillValue', None)
+        data.attrs.pop('Intercept', None)
+        data.attrs.pop('Slope', None)
+
+        data = data.where((data >= data.attrs['valid_range'][0]) &
+                          (data <= data.attrs['valid_range'][1]))
+
+        return data
+
+    def get_area_def(self, key):
+        # Coordination Group for Meteorological Satellites LRIT/HRIT Global Specification
+        # https://www.cgms-info.org/documents/cgms-lrit-hrit-global-specification-(v2-8-of-30-oct-2013).pdf
+        res = key.resolution
+        coff = _COFF_list[_resolution_list.index(res)]
+        loff = _LOFF_list[_resolution_list.index(res)]
+        cfac = _CFAC_list[_resolution_list.index(res)]
+        lfac = _LFAC_list[_resolution_list.index(res)]
+        a = self.file_content['/attr/dEA'] * 1E3  # equator radius (m)
+        b = a * (1 - 1 / self.file_content['/attr/dObRecFlat'])  # polar radius (m)
+        h = self.file_content['/attr/NOMSatHeight']  # the altitude of satellite (m)
+
+        lon_0 = self.file_content['/attr/NOMCenterLon']
+        nlines = self.file_content['/attr/RegLength']
+        ncols = self.file_content['/attr/RegWidth']
+
+        cols = 0
+        left_x = (cols - coff) * (2.**16 / cfac)
+        cols += ncols - 1
+        right_x = (cols - coff) * (2.**16 / cfac)
+
+        lines = self.file_content['/attr/Begin Line Number']
+        upper_y = -(lines - loff) * (2.**16 / lfac)
+        lines = self.file_content['/attr/End Line Number']
+        lower_y = -(lines - loff) * (2.**16 / lfac)
+        area_extent = (np.deg2rad(left_x) * h, np.deg2rad(lower_y) * h,
+                       np.deg2rad(right_x) * h, np.deg2rad(upper_y) * h)
+
+        proj_dict = {'a': float(a),
+                     'b': float(b),
+                     'lon_0': float(lon_0),
+                     'h': float(h),
+                     'proj': 'geos',
+                     'units': 'm',
+                     'sweep': 'y'}
+
+        area = geometry.AreaDefinition(
+            self.filename_info['observation_type'],
+            "AGRI {} area".format(self.filename_info['observation_type']),
+            'FY-4A',
+            proj_dict,
+            ncols,
+            nlines,
+            area_extent)
+
+        self.area = area

```

Comment:
Is this needed? It doesn't seem to be used and my preference would be to not cache the area definitions unless needed.